nanoFirmwareFlasher/Esp32Operations.cs

```diff
@@ -168,9 +168,7 @@ internal class Esp32Operations
                 // need to get deployment address here
                 // length must both be multiples of the SPI flash erase sector size. This is 0x1000 (4096) bytes for supported flash chips.
 
-                var flashPartition = firmware.FlashPartitions.First();
-
-                var fileStream = File.OpenRead(flashPartition.Value);
+                var fileStream = File.OpenRead(firmware.BootloaderPath);
 
                 uint fileLength = (uint)Math.Ceiling((decimal)fileStream.Length / 0x1000) * 0x1000;
 
```

Comment:
Shouldn't it also call `EraseFlashSegment` not from `address` but from the offset of BootloaderPath on firmware? I.e. 0x1000?

Reply:
In the end, that's not necessary, because the addresses being used in the parameters are already the exact addresses in the flash space. (yes, I told you before that those were offset, but that wasn't accurate).